Cannot parse minimum int64 literal (-9223372036854775808)

## Description

The parser cannot handle the minimum int64 value `-9223372036854775808` as a literal.

## Reproduction

```ez
do main() {
    temp min_int int = -9223372036854775808
}
```

## Error

```
error[E2027]: could not parse "9223372036854775808" as integer
```

## Root Cause

The parser parses the number without the negative sign first, then applies negation. Since `9223372036854775808` exceeds int64 max (9223372036854775807), it fails before the negation can be applied.

## Expected Behavior

The minimum int64 value should be parseable as a literal, since it is a valid int64 value.

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -2570,6 +2570,20 @@ func (p *Parser) parsePrefixExpression() Expression {
 		Operator: p.currentToken.Literal,
 	}
 
+	// Special case for minimum int64: -9223372036854775808
+	// This value can't be parsed as a positive int64 first then negated,
+	// because 9223372036854775808 overflows int64.
+	if expression.Operator == "-" && p.peekTokenMatches(INT) {
+		cleanedLiteral := stripUnderscores(p.peekToken.Literal)
+		if cleanedLiteral == "9223372036854775808" {
+			p.nextToken() // consume the integer token
+			return &IntegerValue{
+				Token: p.currentToken,
+				Value: -9223372036854775808, // math.MinInt64
+			}
+		}
+	}
+
 	p.nextToken()
 	expression.Right = p.parseExpression(PREFIX)
 

--- a/pkg/parser/parser_test.go
+++ b/pkg/parser/parser_test.go
@@ -2097,3 +2097,40 @@ do main() {
 		t.Fatal("program is nil")
 	}
 }
+
+func TestMinInt64Literal(t *testing.T) {
+	// Test: minimum int64 value should parse correctly
+	// Fixes #346
+	input := `temp x int = -9223372036854775808`
+	l := NewLexer(input)
+	p := NewWithSource(l, input, "test.ez")
+	program := p.ParseProgram()
+
+	if p.EZErrors().HasErrors() {
+		t.Errorf("unexpected parser errors: %v", p.Errors())
+	}
+
+	if program == nil {
+		t.Fatal("program is nil")
+	}
+
+	// Should have one statement (variable declaration)
+	if len(program.Statements) != 1 {
+		t.Fatalf("expected 1 statement, got %d", len(program.Statements))
+	}
+
+	// The value should be an IntegerValue with min int64
+	varDecl, ok := program.Statements[0].(*VariableDeclaration)
+	if !ok {
+		t.Fatalf("expected VariableDeclaration, got %T", program.Statements[0])
+	}
+
+	intVal, ok := varDecl.Value.(*IntegerValue)
+	if !ok {
+		t.Fatalf("expected IntegerValue, got %T", varDecl.Value)
+	}
+
+	if intVal.Value != -9223372036854775808 {
+		t.Errorf("expected -9223372036854775808, got %d", intVal.Value)
+	}
+}

